File: api/stash/definitions/stash.ts
export type InventoryType = "stash" | "trunk" | "glovebox";

export interface InventoryRef {
    type: InventoryType;
    id: number | string;
}

export interface ServerStashProvider {
    addItem(inventoryRef: InventoryRef, itemName: string, amount: number): boolean;
    removeItem(inventoryRef: InventoryRef, itemName: string, amount: number): boolean;
    getItemCount(inventoryRef: InventoryRef, itemName: string): number;
    hasItem(inventoryRef: InventoryRef, itemName: string, amount: number): boolean;
}

File: api/stash/definitions/stash.lua
---@alias InventoryType "stash" | "trunk" | "glovebox"

---@class InventoryRef
---@field type InventoryType
---@field id integer | string

---@class ServerStashProvider
---@field addItem fun(inventoryRef: InventoryRef, itemName: string, amount: number): boolean
---@field removeItem fun(inventoryRef: InventoryRef, itemName: string, amount: number): boolean
---@field getItemCount fun(inventoryRef: InventoryRef, itemName: string): number
---@field hasItem fun(inventoryRef: InventoryRef, itemName: string, amount: number): boolean
---@field canAddItem fun(inventoryRef: InventoryRef, itemName: string, amount: number): boolean
